src/Map/Map.h

```diff
@@ -34,7 +34,7 @@ class Map
 	Map();
 
 	static Map ReadMap(std::string filename);
-	static Map ReadMap(Stream::Reader& seekableReader);
+	static Map ReadMap(Stream::SeekableReader& seekableReader);
```

Comment:
Would be nice to not need this change, though I can see why we might want to do this for now. Perhaps at some point we could factor out the common part into a method. The `if` controlled part could then be unconditionally called from the `ReadSavedGame` method.

Reply:
Good idea. Since we are peppering the code with if statements anyways, might as well separate the code.